sagemaker-clarify/bias-and-explainability/student_performance_clarify.ipynb

```diff
@@ -0,0 +1,860 @@
+    ")"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "#### Define Data Configuration"
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "bias_pretrain_report_output_path = \"s3://{}/{}/clarify-pretrain-bias\".format(bucket, prefix)\n",
+    "\n",
+    "bias_data_config = clarify.DataConfig(\n",
+    "    s3_data_input_path=train_data_s3_path,\n",
+    "    s3_output_path=bias_pretrain_report_output_path,\n",
+    "    label=\"FinalGrade\",\n",
+    "    headers=train_data.columns.to_list(),\n",
+    "    dataset_type=\"text/csv\",\n",
+    ")"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "#### Define Bias Configuration\n",
+    "\n",
+    "Now we can run pre-bias training over the favorable values, or the top final grades."
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "bias_config = clarify.BiasConfig(\n",
+    "    label_values_or_threshold=[15, 16, 17, 18, 19, 20], facet_name=\"Gender\"\n",
+    ")"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "#### Run Pre-training Bias Job\n",
+    "\n",
+    "Run the SageMaker Analyzer job - this takes about 5 minutes."
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "%time\n",
+    "\n",
+    "clarify_processor.run_pre_training_bias(\n",
+    "    data_config=bias_data_config, data_bias_config=bias_config, methods=\"all\"\n",
+    ")"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Once the report has been generated, we can inspect what our values are looking like."
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "from IPython.core.display import HTML\n",
+    "\n",
+    "url = \"https://s3.console.aws.amazon.com/s3/buckets/{}?region={}&prefix={}/clarify-bias/&showversions=false\".format(\n",
+    "    bucket, session.boto_region_name, prefix\n",
+    ")\n",
+    "HTML(\n",
+    "    '<a target=\"_blank\" href=\"{}\">Click here to view pre-training bias reports in S3</a>'.format(\n",
+    "        url\n",
+    "    )\n",
+    ")\n",
+    "from sagemaker.s3 import S3Downloader\n",
+    "\n",
+    "local_databias_report_path = \"./pretraining_bias_reports\"\n",
+    "\n",
+    "S3Downloader.download(\n",
+    "    local_path=local_databias_report_path,\n",
+    "    s3_uri=bias_pretrain_report_output_path,\n",
+    "    sagemaker_session=session,\n",
+    ")"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Inspect the Pretraining Bias Report\n",
+    "\n",
+    "You can view the report [here](./pretraining_bias_reports/report.ipynb)."
```

Comment:
Confirming that the report notebooks will be generated and downloaded from running the notebook?

Reply:
Yes, the reports are locally available.

`
.
├── README.md
├── clarify_explainability_arch.png
├── explainability_reports
│   ├── analysis.json
│   ├── analysis_config.json
│   ├── explanations_shap
│   │   ├── baseline.csv
│   │   └── out.csv
│   ├── report.html
│   ├── report.ipynb
│   └── report.pdf
├── posttraining_bias_reports
│   ├── analysis.json
│   ├── analysis_config.json
│   ├── report.html
│   ├── report.ipynb
│   └── report.pdf
├── pretraining_bias_reports
│   ├── analysis.json
│   ├── analysis_config.json
│   ├── report.html
│   ├── report.ipynb
│   └── report.pdf
├── student_performance_clarify.ipynb
├── train_data.csv
└── validation_data.csv

4 directories, 22 files
`